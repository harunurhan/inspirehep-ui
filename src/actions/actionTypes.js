export const SEARCH_REQUEST = 'SEARCH_REQUEST';
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_ERROR = 'SEARCH_ERROR';

export const LITERATURE_REQUEST = 'LITERATURE_REQUEST';
export const LITERATURE_SUCCESS = 'LITERATURE_SUCCESS';
export const LITERATURE_ERROR = 'LITERATURE_ERROR';
export const LITERATURE_REFERENCES_REQUEST = 'LITERATURE_REFERENCES_REQUEST';
export const LITERATURE_REFERENCES_SUCCESS = 'LITERATURE_REFERENCES_SUCCESS';
export const LITERATURE_REFERENCES_ERROR = 'LITERATURE_REFERENCES_ERROR';

export const EXCEPTIONS_REQUEST = 'EXCEPTIONS_REQUEST';
export const EXCEPTIONS_SUCCESS = 'EXCEPTIONS_SUCCESS';
export const EXCEPTIONS_ERROR = 'EXCEPTIONS_ERROR';
