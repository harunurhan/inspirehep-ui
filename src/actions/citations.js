import {
  CITATIONS_ERROR,
  CITATIONS_SUCCESS,
  CITATIONS_REQUEST,
} from './actionTypes';

function fetching() {
  return {
    type: CITATIONS_REQUEST,
  };
}

function fetchSuccess(result) {
  return {
    type: CITATIONS_SUCCESS,
    payload: result,
  };
}

function fetchError(error) {
  return {
    type: CITATIONS_ERROR,
    payload: error,
  };
}

export default function fetch(pidType, recordId, paginationOptions) {
  return async (dispatch, getState, http) => {
    const { page, pageSize } = paginationOptions;
    dispatch(fetching());
    try {
      const citationsApiUrl = `/${pidType}/${recordId}/citations?page=${page}&size=${pageSize}`;
      const response = await http.get(citationsApiUrl);
      dispatch(fetchSuccess(response.data));
    } catch (error) {
      dispatch(fetchError(error.response && error.response.data));
    }
  };
}
