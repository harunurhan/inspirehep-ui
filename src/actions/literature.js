import {
  LITERATURE_ERROR,
  LITERATURE_REQUEST,
  LITERATURE_SUCCESS,
  LITERATURE_REFERENCES_ERROR,
  LITERATURE_REFERENCES_REQUEST,
  LITERATURE_REFERENCES_SUCCESS,
  LITERATURE_AUTHORS_ERROR,
  LITERATURE_AUTHORS_REQUEST,
  LITERATURE_AUTHORS_SUCCESS,
} from './actionTypes';
import { UI_SERIALIZER_REQUEST_OPTIONS } from '../common/http';
import { httpErrorToActionPayload } from '../common/utils';

function fetchingLiterature(recordId) {
  return {
    type: LITERATURE_REQUEST,
    payload: { recordId },
  };
}

function fetchLiteratureSuccess(result) {
  return {
    type: LITERATURE_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureError(error) {
  return {
    type: LITERATURE_ERROR,
    payload: error,
    meta: { redirectableError: true },
  };
}

function fetchingLiteratureReferences() {
  return {
    type: LITERATURE_REFERENCES_REQUEST,
  };
}

function fetchLiteratureReferencesSuccess(result) {
  return {
    type: LITERATURE_REFERENCES_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureReferencesError(error) {
  return {
    type: LITERATURE_REFERENCES_ERROR,
    payload: error,
  };
}

function fetchingLiteratureAuthors() {
  return {
    type: LITERATURE_AUTHORS_REQUEST,
  };
}

function fetchLiteratureAuthorsSuccess(result) {
  return {
    type: LITERATURE_AUTHORS_SUCCESS,
    payload: result,
  };
}

function fetchLiteratureAuthorsError() {
  return {
    type: LITERATURE_AUTHORS_ERROR,
  };
}

export function fetchLiterature(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiterature(recordId));
    try {
      const response = await http.get(
        `/literature/${recordId}`,
        UI_SERIALIZER_REQUEST_OPTIONS
      );
      dispatch(fetchLiteratureSuccess(response.data));
    } catch (error) {
      const payload = httpErrorToActionPayload(error);
      dispatch(fetchLiteratureError(payload));
    }
  };
}

export function fetchLiteratureReferences(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiteratureReferences());
    try {
      const response = await http.get(`/literature/${recordId}/references`);
      dispatch(fetchLiteratureReferencesSuccess(response.data));
    } catch (error) {
      const payload = httpErrorToActionPayload(error);
      dispatch(fetchLiteratureReferencesError(payload));
    }
  };
}

export function fetchLiteratureAuthors(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingLiteratureAuthors());
    try {
      const response = await http.get(`/literature/${recordId}/authors`);
      dispatch(fetchLiteratureAuthorsSuccess(response.data));
    } catch (error) {
      dispatch(fetchLiteratureAuthorsError());
    }
  };
}
