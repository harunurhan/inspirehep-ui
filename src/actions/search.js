import { push } from 'react-router-redux';
import { stringify } from 'qs';

import {
  SEARCH_REQUEST,
  SEARCH_ERROR,
  SEARCH_SUCCESS,
  CHANGE_SEARCH_SCOPE,
  SEARCH_AGGREGATIONS_REQUEST,
  SEARCH_AGGREGATIONS_SUCCESS,
  SEARCH_AGGREGATIONS_ERROR,
} from './actionTypes';

export function changeSearchScope(scope) {
  return {
    type: CHANGE_SEARCH_SCOPE,
    payload: scope,
  };
}

function searching() {
  return {
    type: SEARCH_REQUEST,
  };
}

function searchSuccess(result) {
  return {
    type: SEARCH_SUCCESS,
    payload: result,
  };
}

function searchError(error) {
  return {
    type: SEARCH_ERROR,
    payload: error,
  };
}

export function searchForCurrentLocation() {
  return async (dispatch, getState, http) => {
    const { location } = getState().router;
    dispatch(searching());
    const url = `${location.pathname}${location.search}`;
    try {
      const response = await http.get(url, {
        headers: {
          Accept: 'application/vnd+inspire.record.ui+json',
        },
      });
      dispatch(searchSuccess(response.data));
    } catch (error) {
      dispatch(searchError(error.data));
    }
  };
}

function fetchingSearchAggregations() {
  return {
    type: SEARCH_AGGREGATIONS_REQUEST,
  };
}

function searchAggregationsSuccess(result) {
  return {
    type: SEARCH_AGGREGATIONS_SUCCESS,
    payload: result,
  };
}

function searchAggregationsError(error) {
  return {
    type: SEARCH_AGGREGATIONS_ERROR,
    payload: error,
  };
}

export function fetchSearchAggregationsForCurrentLocation() {
  return async (dispatch, getState, http) => {
    const { location } = getState().router;
    dispatch(fetchingSearchAggregations());
    const url = `${location.pathname}/facets${location.search}`;
    try {
      const response = await http.get(url, {
        headers: {
          Accept: 'application/vnd+inspire.record.ui+json',
        },
      });
      dispatch(searchAggregationsSuccess(response.data));
    } catch (error) {
      dispatch(searchAggregationsError(error.response && error.response.data));
    }
  };
}

function getSearchUrl(state, query) {
  const pathname = state.search.getIn(['scope', 'pathname']);
  const queryString = stringify(query, { indices: false });
  return `/${pathname}?${queryString}`;
}

function appendQuery(state, query, excludeLocationQuery) {
  const baseQuery = state.search.getIn(['scope', 'query']).toJS();
  const locationQuery = excludeLocationQuery ? {} : state.router.location.query;

  if (query && locationQuery.page !== undefined) {
    locationQuery.page = 1;
  }

  return {
    ...baseQuery,
    ...locationQuery,
    ...query,
  };
}

export function pushQueryToLocation(query, clearLocationQuery = false) {
  return async (dispatch, getState) => {
    const state = getState();
    const newQuery = appendQuery(state, query, clearLocationQuery);
    const url = getSearchUrl(state, newQuery);
    if (Object.keys(newQuery).length > 0) {
      dispatch(push(url));
    }
  };
}
