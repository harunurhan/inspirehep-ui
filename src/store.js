import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
/* eslint-disable import/no-extraneous-dependencies */

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import persistUserStateMiddleware, {
  reHydrateRootStateWithUser,
} from './middlewares/persistUserState';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const getMiddleware = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(
      reduxRouterMiddleware,
      queryParamsParserMiddleware,
      persistUserStateMiddleware,
      thunkMiddleware
    );
  }
  return applyMiddleware(
    reduxRouterMiddleware,
    queryParamsParserMiddleware,
    persistUserStateMiddleware,
    thunkMiddleware,
    createLogger()
  );
};

export const store = createStore(
  reducers,
  reHydrateRootStateWithUser(),
  composeWithDevTools(getMiddleware())
);
