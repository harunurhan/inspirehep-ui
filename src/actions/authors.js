import { AUTHOR_REQUEST, AUTHOR_SUCCESS, AUTHOR_ERROR } from './actionTypes';

function fetchingAuthor(recordId) {
  return {
    type: AUTHOR_REQUEST,
    payload: { recordId },
  };
}

function fetchAuthorSuccess(result) {
  return {
    type: AUTHOR_SUCCESS,
    payload: result,
  };
}

function fetchAuthorError(error) {
  return {
    type: AUTHOR_ERROR,
    payload: error,
  };
}

export default function fetchAuthor(recordId) {
  return async (dispatch, getState, http) => {
    dispatch(fetchingAuthor(recordId));
    try {
      const response = await http.get(`/authors/${recordId}`, {
        headers: {
          Accept: 'application/vnd+inspire.record.ui+json',
        },
      });
      dispatch(fetchAuthorSuccess(response.data));
    } catch (error) {
      dispatch(fetchAuthorError(error.response && error.response.data));
    }
  };
}
