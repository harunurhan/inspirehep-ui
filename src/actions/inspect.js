import { INSPECT_REQUEST, INSPECT_SUCCESS, INSPECT_ERROR } from './actionTypes';

function fetching(id) {
  return {
    type: INSPECT_REQUEST,
    payload: { id },
  };
}

function fetchSuccess(result) {
  return {
    type: INSPECT_SUCCESS,
    payload: result,
  };
}

function fetchError(error) {
  return {
    type: INSPECT_ERROR,
    payload: error,
  };
}

export default function fetch(id) {
  return async (dispatch, getState, http) => {
    dispatch(fetching(id));
    try {
      const response = await http.get(`/workflows/inspect_merge/${id}`);
      dispatch(fetchSuccess(response.data));
    } catch (error) {
      dispatch(fetchError(error.data));
    }
  };
}
