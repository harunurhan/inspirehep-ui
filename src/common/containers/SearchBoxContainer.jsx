import { connect } from 'react-redux';

import SearchBox from '../components/SearchBox';
import { pushQueryToLocation } from '../../actions/search';

const stateToProps = state => ({
  value: state.router.location.query.q,
  searchScopeName: state.search.getIn(['scope', 'name']),
});

export const dispatchToProps = dispatch => ({
  onSearch(value) {
    dispatch(pushQueryToLocation({ q: value }, true));
  },
});

export default connect(stateToProps, dispatchToProps)(SearchBox);
