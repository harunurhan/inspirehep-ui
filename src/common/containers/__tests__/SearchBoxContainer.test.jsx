import React from 'react';
import { shallow } from 'enzyme';
import { CALL_HISTORY_METHOD } from 'react-router-redux';

import { getStoreWithState, getStore } from '../../../fixtures/store';
import { SEARCH_REQUEST } from '../../../actions/actionTypes';
import SearchBoxContainer, { dispatchToProps } from '../SearchBoxContainer';

describe('SearchBoxContainer', () => {
  it('renders initial state with initial url query q param', () => {
    const store = getStoreWithState({
      router: { location: { query: { q: 'test' } } },
    });
    const wrapper = shallow(<SearchBoxContainer store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches search onSearch', () => {
    const store = getStore();
    const value = 'test';
    const props = dispatchToProps(store.dispatch);
    props.onSearch(value);
    const actions = store.getActions();
    const searchRequestAction = actions.find(
      action => action.type === SEARCH_REQUEST
    );
    expect(searchRequestAction).toBeDefined();
    expect(searchRequestAction.payload).toEqual({ q: value });
  });

  it('clears existing search params onSearch', () => {
    const store = getStoreWithState({
      router: { location: { query: { another: 'value' } } },
    });
    const value = 'test';
    const props = dispatchToProps(store.dispatch);
    props.onSearch(value);
    const actions = store.getActions();
    const searchUrlPushAction = actions.find(
      action => action.type === CALL_HISTORY_METHOD
    );
    expect(searchUrlPushAction).toBeDefined();
    const searchUrl = searchUrlPushAction.payload.args[0];
    expect(searchUrl).not.toMatch('another');
  });
});
