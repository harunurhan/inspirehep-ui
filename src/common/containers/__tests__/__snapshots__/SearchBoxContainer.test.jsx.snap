// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxContainer renders initial state with initial url query q param 1`] = `
<Search
  addonBefore="literature"
  defaultValue="test"
  enterButton={true}
  inputPrefixCls="ant-input"
  onSearch={[Function]}
  placeholder={null}
  prefixCls="ant-input-search"
  style={
    Object {
      "verticalAlign": "middle",
    }
  }
/>
`;
